viewer/app/index.html

```diff
@@ -43,7 +43,7 @@ <h1 class="viewer-placeholder__heading">Lighthouse Report Viewer</h1>
 
 <input id="hidden-file-input" type="file" hidden accept="application/json">
 
-<script src="src/bundled.js"></script>
```

Comment:
Just spent the better part of an afternoon/evening on this. `bundled.js` is created as an ESM but we weren't loading it with `type="module"`:
https://github.com/GoogleChrome/lighthouse/blob/abb88b36c7f95af11a1f707bbd63c6616bfaad0d/build/gh-pages-app.js#L159
It wasn't a problem before because `bundle.js` didn't use any static `import`/`export` statements. That changed when we added the flow renderer to the bundle.

The flow renderer imports `shated/localization/format.js`, which is also used in the dynamically imported `i18n-module.js`. So now `format.js` is used in the initially loaded code and the dynamically loaded i18n bundle.

This creates a new `main-xxxx.js` bundle that is imported in `bundle.js` and the i18n bundle. At this point the browser gets mad because because `bundle.js` is using a static `import` even though the script tag didn't have `type="module"`.

Is there a reason `type="module"` was not added originally? If so, we might get around this by loading the flow assets dynamically.

Reply:
background: https://github.com/GoogleChrome/lighthouse/pull/10148#discussion_r721686540

tldr: adding `type=module` should be fine

my main concern previously was that concating multiple esm bundles would not work, and so I thought best to not mark the script element as a module so we did not accidentally rely on that. however on second thought I do think concat'ing multiple es module bundles (which we don't do now...) would be fine because imports are simply hoisted. 2 months ago I hadn't realized that, thinking they would error.

nice to know that rollup with extract these out into a separate file and load it. maybe we should add a link preload element for it though? concerned about the delayed loading here. Is there a way to enumerate all the static imports from the main bundle and emit link elements for it inside `build-gh-pages-app.js`?

Reply:
> Is there a way to enumerate all the static imports from the main bundle and emit link elements for it inside build-gh-pages-app.js?

I think so looks like each output chunk has a convenient `isDynamicEntry` flag.